Content.Server/DeviceNetwork/Systems/NetworkConfiguratorSystem.cs

```diff
@@ -0,0 +1,284 @@
+﻿using Content.Server.DeviceNetwork.Components;
+using Content.Server.UserInterface;
+using Content.Shared.Access.Components;
+using Content.Shared.Access.Systems;
+using Content.Shared.Database;
+using Content.Shared.DeviceNetwork;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Movement;
+using Content.Shared.Popups;
+using Content.Shared.Verbs;
+using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+
+namespace Content.Server.DeviceNetwork.Systems;
+
+[UsedImplicitly]
+public sealed class NetworkConfiguratorSystem : EntitySystem
+{
+    [Dependency] private readonly DeviceListSystem _deviceListSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+    [Dependency] private readonly AccessReaderSystem _accessSystem = default!;
+    [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        //Interaction
+        SubscribeLocalEvent<NetworkConfiguratorComponent, AfterInteractEvent>((uid, component, args) => OnUsed(uid, component, args.Target, args.User, args.CanReach)); //TODO: Replace with utility verb?
+
+        //Verbs
+        SubscribeLocalEvent<NetworkConfiguratorComponent, GetVerbsEvent<UtilityVerb>>(OnAddInteractVerb);
+        SubscribeLocalEvent<DeviceNetworkComponent, GetVerbsEvent<AlternativeVerb>>(OnAddAlternativeSaveDeviceVerb);
+
+        //UI
+        SubscribeLocalEvent<NetworkConfiguratorComponent, BoundUIClosedEvent>(OnUiClosed);
+        SubscribeLocalEvent<NetworkConfiguratorComponent, NetworkConfiguratorRemoveDeviceMessage>(OnRemoveDevice);
+        SubscribeLocalEvent<NetworkConfiguratorComponent, NetworkConfiguratorClearDevicesMessage>(OnClearDevice);
+        SubscribeLocalEvent<NetworkConfiguratorComponent, NetworkConfiguratorButtonPressedMessage>(OnConfigButtonPressed);
+
+        SubscribeLocalEvent<DeviceListComponent, ComponentRemove>(OnComponentRemoved);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var component in EntityManager.EntityQuery<NetworkConfiguratorComponent>())
```

Comment:
The new convention for this sort of thing is to add a new `ActiveNetworkConfiguratorComponent` and enumerate over that, rather than enumerating over every entity with a `NetworkConfiguratorComponent` every tick. Though honestly this sort of range/obstruction check that involves a third entity that's not the player or BUI provider should probably just be supported by BUIs in the first place. So its probably fine to just leave as is and remove this whenever that functionality gets added.

Reply:
Yea I also think that should be supported by BUIs so I will remove this when that gets added.